tests/telemetry/test_telemetry.py

```diff
@@ -149,6 +149,11 @@ def test_telemetry_queue_buffer_cnt(duthosts, enum_rand_one_per_hwsku_hostname,
     if duthost.is_supervisor_node():
         pytest.skip(
             "Skipping test as no Ethernet0 frontpanel port on supervisor")
+
+    # If switch type is voq, switch will not support creating buffer queue from config db.
+    if duthost.facts['switch_type'] == 'voq':
```

Comment:
move this check to condition marker yaml file

Reply:
@arlakshm  @vmittal-msft  Currently the basic facts does not contain switch_type for condition marker check. If you want the check to be there we need to add code to generate the basic fact containing switch_type. Let me know if you still want to go that way then I can do some research on how to make basic facts have that. 

Current available basic facts that can be used in conditional skip (sample dump from the test output):
{
  "topo_type": "t2",
  "topo_name": "t2_ixre_chassis14",
  "testbed": "ixre-chassis14-t2",
  "asic_subtype": "broadcom-dnx",
  "asic_type": "broadcom",
  "branch": "HEAD",
  "build_date": "Wed Sep 25 15:57:47 UTC 2024",
  "build_number": "832829-nokia-master",
  "build_version": "HEAD.832829-nokia-master-bf8e2c9a7",
  "built_by": "gitlab-runner@wfrv-sonicbld06",
  "commit_id": "bf8e2c9a7",
  "debian_version": "12.7",
  "feature_status": {
  },
  "hwsku": "Nokia-IXR7250E-36x400G",
  "is_multi_asic": true,
  "is_supervisor": false,
  "kernel_version": "6.1.0",
  "libswsscommon": "1.0.0",
  "num_asic": 2,
  "platform": "x86_64-nokia_ixr7250e_36x400g-r0",
  "release": "master",
  "secure_boot_image": "no",
  "sonic_os_version": 12,
  "sonic_utilities": 1.2,
  "asic_gen": "unknown",
   ..............................
}

Reply:
OK. Add support for switch_type on base facts and move check to conditional mark. Please review.